ios/FluentUI/Vnext/Button/Button.swift

```diff
@@ -3,21 +3,101 @@
 //  Licensed under the MIT License.
 //
 
-import Combine
-import UIKit
 import SwiftUI
+import UIKit
+
+/// View that represents the button
+public struct MSFButtonView: View {
+    @Environment(\.theme) var theme: FluentUIStyle
+    @Environment(\.windowProvider) var windowProvider: FluentUIWindowProvider?
+    @ObservedObject var tokens: MSFButtonTokens
+    @ObservedObject var state: MSFButtonStateImpl
+    var action: () -> Void
+
+    public init(action: @escaping () -> Void,
+                style: MSFButtonStyle,
+                size: MSFButtonSize,
+                image: UIImage? = nil,
+                text: String? = nil) {
+        let state = MSFButtonStateImpl(style: style, size: size)
+        state.text = text
+        state.image = image
+        self.state = state
+        self.tokens = state.tokens
+        self.action = action
+    }
 
-/// Properties available to customize the state of the button
-@objc public class MSFButtonState: NSObject, ObservableObject {
-    @objc @Published public var image: UIImage?
-    @objc @Published public var isDisabled: Bool = false
-    @objc @Published public var text: String?
+    public var body: some View {
+        Button(action: action, label: {})
+            .buttonStyle(MSFButtonViewButtonStyle(tokens: tokens,
+                                                  state: state))
+            .disabled(state.isDisabled)
+            .frame(maxWidth: .infinity)
+            .designTokens(tokens,
+                          from: theme,
+                          with: windowProvider)
+    }
+}
+
+/// Properties that can be used to customize the appearance of the Button.
+@objc public protocol MSFButtonState {
```

Comment:
Nit: other finalized vNext classes have put the state protocol above the view struct. Nice opportunity for alignment/readability.